exec/java-exec/src/main/java/org/apache/drill/exec/rpc/user/security/PamUserAuthenticator.java

```diff
@@ -59,7 +59,8 @@ public void authenticate(String user, String password) throws UserAuthentication
     for (String pamProfile : profiles) {
       Pam pam = new Pam(pamProfile);
       if (!pam.authenticateSuccessful(user, password)) {
-        throw new UserAuthenticationException(String.format("PAM profile '%s' validation failed", pamProfile));
+        throw new UserAuthenticationException(String.format("PAM profile '%s' validation failed for user %s",
```

Comment:
The module above throws an exception. Here we get a `boolean`. Does this mean that the exception was mapped to `false`, then mapped back to an exception?

Reply:
No the exceptions thrown here is directly mapped to Sasl Exception and sent back to client with AUTH_FAILED status.